File: packages/shared-types/src/job.ts
export const jobTypes = ["transcribe", "summarize", "export"] as const;
export const jobStatuses = [
  "queued",
  "processing",
  "completed",
  "failed",
] as const;

export type JobType = (typeof jobTypes)[number];
export type JobStatus = (typeof jobStatuses)[number];

export interface JobRecord {
  id: string;
  meetingId: string;
  jobType: JobType;
  status: JobStatus;
  progress: number;
  errorMessage?: string | null;
}

File: apps/api/app/schemas/job.py
from typing import Literal

from pydantic import BaseModel


JobType = Literal["transcribe", "summarize", "export"]
JobStatus = Literal["queued", "processing", "completed", "failed"]


class JobStatusResponse(BaseModel):
    id: str
    meeting_id: str
    job_type: JobType
    status: JobStatus
    progress: int
